scipy/stats/_continuous_distns.py

```diff
@@ -3492,7 +3492,7 @@ def _argcheck(self, a):
             # shape parameter, so warn the user.
             message = ('The shape parameter of the erlang distribution '
                        f'has been given a non-integer value {a!r}.')
-            warnings.warn(message, RuntimeWarning)
+            warnings.warn(message, RuntimeWarning, stacklevel=2)
```

Comment:
Yeah I think `stacklevel=2` is good for public functions, but since private functions are never called directly by the user, I think `stacklevel=3` is a good minimum for private functions. For instance,
```python3
from scipy.stats import erlang; erlang.pdf(1, 1.4)
```
With `stacklevel=2`:
```
/Users/matthaberland/Desktop/scipy/scipy/stats/_distn_infrastructure.py:1987: RuntimeWarning: The shape parameter of the erlang distribution has been given a non-integer value array(1.4).
  cond0 = self._argcheck(*args) & (scale > 0)
```
With `stacklevel=3`:
```
<ipython-input-2-f4fb45a9f0ad>:1: RuntimeWarning: The shape parameter of the erlang distribution has been given a non-integer value array(1.4).
  from scipy.stats import erlang; erlang.pdf(1, 1.4)
```
(which is what we want).